### ISSUE TITLE
Decorated property not supported

### ISSUE THREAD

**Issue by liu-yabin** [Microsoft/pyright#92]
```python
class MyClass:
    def __init__(self, prop1: int = 1):
        self._prop1: int = prop1

    @property
    def prop1(self) -> int:
        return self._prop1

    @prop1.setter
    def prop1(self, prop1: int):
        self._prop1 = prop1


obj1 = MyClass()
obj1.prop1 = "abc"
print(obj1.prop1)
```
pyright is a good tools to use in vscode. But I found  it don't supported decorated property.
In the code above, I hope the property `prop1` can only be  a value of `int` type. But when I assigned it with a value of `abc`(`string` type), pyright didn't give me a error message.
I think this feature is very common, could you possibly solve the problem?

my python version is *python 3.7.1 64-bit*.

![EnhAPI.png](https://s2.ax1x.com/2019/04/26/EnhAPI.png)

**Comment by erictraut** [comment]
Thanks for the request. Yes, this is a type checking feature that is not yet implemented in pyright. More generally, it doesn't check assignments to any object that has a `__set__` method. Properties are just the most common usage.

**Comment by erictraut** [comment]
This is now fixed in 1.0.19, which I just published.
